ESP32/lib/wifi/wifi_handler.h

```diff
@@ -59,15 +57,16 @@ class Wifi_handler {
   // this function is used that a class member function can be used as event
   // handler function
   static void event_handler_static(void *event_handler_arg,
-                                   esp_event_base_t event_base,
-                                   int32_t event_id, void *event_data);
+                                   const esp_event_base_t event_base,
+                                   const int32_t event_id, void *event_data);
 
-  const std::string tag;
-  std::string m_ssid;
-  std::string m_password;
+ private:
+  const std::string m_tag = "wifi_handler";
+  std::string m_ssid = "";
+  std::string m_password = "";
   uint8_t m_retry_num = 0;
   bool m_connecting = false;
-  esp_netif_t *wifi_netif = NULL;
+  esp_netif_t *m_wifi_netif = NULL;
 };
```

Comment:
You initialized `m_tag` twice, once here and once as init list at the constructor. So remove it here and init all the other variables also at the constructor

Reply:
Yes double initialisation to fixed value is not good.
I would still keep the defaults in the header file, so values are always initialised. 

The c++ guidelines also seems to favour the initialisation in the class (C45 and C48).
https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#c45-dont-define-a-default-constructor-that-only-initializes-data-members-use-default-member-initializers-instead

C45 also says to remove the constructor altogether (and use the default constructor), but I kinda like to keep it, to simply have it if I want to use it and I think it does not change anything if it is there (I hope 😅).

Reply:
We will open a discussion about that and handle this in a separate issue